src/uwtools/resources/rocoto.jsonschema

```diff
@@ -3,68 +3,250 @@
         "dateTimePattern": {
             "type": "string",
             "pattern": "^(\\d{12}) (\\d{12}) (\\d{2}):(\\d{2}):(\\d{2})$"
-        }
-    },
-    "type": "object",
-    "properties": {
-        "workflow": {
-            "type": "object",
-            "properties": {
-                "attrs": {
+        },
+        "TimePattern": {
+            "type": "string",
+            "pattern": "^(\\d{2}):(\\d{2}):(\\d{2})$$"
+        },
+        "DatePattern": {
+            "type": "integer",
+            "format": "string",
+            "pattern": "^(\\d{12}) (\\d{12})$"
+        },
+        "taskPattern": {
+            "patternProperties": {
+                "^task_.*$": {
                     "type": "object",
                     "properties": {
-                        "realtime": {
-                            "type": "boolean"
-                        },
-                        "scheduler": {
-                            "type": "string",
-                            "enum": [
-                                "sge",
-                                "lsf",
-                                "lsfcray",
-                                "ll",
-                                "torque",
-                                "moabtorque",
-                                "moab",
-                                "pbspro",
-                                "slurm",
-                                "cobalt"
-                            ]
+                        "attrs": {
+                            "maxtries": {
+                                "type": "string",
+                                "format": "number",
+                                "minimum": 0
+                            },
+                            "throttle": {
+                                "type": "integer",
+                                "minimum": 0
+                            },
+                            "cycledefs": {
+                                "type": "string"
+                            },
+                            "final": {
+                                "type": "string",
+                                "enum": [
```

Comment:
Not sure what you mean here, could you expand a bit more? A bool for as long as attrs is an `object` then it's fine? Something I thought about looking at this is, is the yaml below valid if we do not have an `attrs` section altogether?

```
workflow:
    attrs:
      realtime: false
      scheduler: slurm
      cyclethrottle: 5
    cycledefs:
      howdy: 
        - 202209290000 202209300000 06:00:00
        - 202210200000 202310201800 02:00:00
      groupname: 
        - 202209290000 202209300000 06:00:00
      groupname2: 
        - 202209290000 202209300000 06:00:00
    entities:
      ACCOUNT: myaccount
      FOO: test.log
    log: /some/path/to/&FOO;
    tasks:
      task_hello:
        account: "&ACCOUNT;"
        command: "echo hello $person"
        walltime: 00:01:00
        memory: 512M
        deadline: 202209290000
        envar:
          name: siri
        cores: 3
        native: hello
```
Meaning should I be removing this required for `task_*` to include `attrs` at all? If `attrs` is left blank currently we get a type object none error. Seeing as everything is optional in `attrs` I'm guessing we want to remove this requirement? Only for `task_*` not for `workflow` tag

Reply:
I think it works just fine if we make `attrs` optional for a `task` entry.

For the boolean part, I was suggesting that instead of a variety of true/false type strings, we just require it to be something that YAML will interpret as a bool. If we do that, we can clean up this enum section to something like this:

```
final: {
    "type": "boolean"
}
```